### PR TITLE
[Doc] Revamp ray core design patterns doc [12/n]: global variables

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
- Copy edit: minor fix for wordings.
- Move the code to doc_code

## Related issue number
#27048

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/patterns/global-variables.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Anti-pattern: Accessing or modifying global variables in tasks and actors
```
Comment (jjyao) [inline doc/source/ray-core/patterns/global-variables.rst:1]: @stephanie-wang: do you have suggestions to the title to make it more like others: doing something causes/harms xxx

Comment (stephanie-wang) [reply doc/source/ray-core/patterns/global-variables.rst:1]: Hmm I guess we don't have a good "consequence" since it's really just unintuitive effect compared to single-process Python.

But how about updating to be more specific?
"Using global variables to share state between tasks and actors"

File: doc/source/ray-core/patterns/global-variables.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+=========================================================================
+
+**TLDR:** Don't access or modify global variables in remote functions and classes. Instead, encapsulate the global variables into an actor's instance variables.
```
Comment (stephanie-wang) [inline doc/source/ray-core/patterns/global-variables.rst:4]:
```suggestion
**TLDR:** Don't use global variables to share state with tasks and actors. Instead, encapsulate the global variables in an actor and pass the actor handle to other tasks and actors.
```

File: doc/source/ray-core/patterns/global-variables.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Ray drivers, tasks and actors are running in
+different processes that don’t share the same address space with each other.
+That says if you modify global variables
+in one process, changes are not reflected in other processes.
```
Comment (stephanie-wang) [inline doc/source/ray-core/patterns/global-variables.rst:9]:
```suggestion
Ray drivers, tasks and actors are running in
different processes, so they don’t share the same address space.
This means that if you modify global variables
in one process, changes are not reflected in other processes.
```

**Review by stephanie-wang — APPROVED** [review]
See suggestion about title change.
